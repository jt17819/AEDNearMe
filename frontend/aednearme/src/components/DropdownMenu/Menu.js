import * as React from 'react'; 

import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import { Link } from 'react-router-dom';

export const MyMenu = (props) => {
    const [anchorEl, setAnchorEl] = React.useState(null);
    const open = Boolean(anchorEl);
    const handleClick = (event) => {
        setAnchorEl(event.currentTarget);
    };
    const handleClose = () => {
        setAnchorEl(null);
    };

    return (
        <div>
            <span
                id="basic-button"
                aria-controls={open ? 'basic-menu' : undefined}
                aria-haspopup="true"
                aria-expanded={open ? 'true' : undefined}
                onClick={handleClick}
                style={{
                    color: "#fff",
                    fontSize: '20px',
                    cursor: 'pointer',
                    marginLeft: '20px',
                    fontWeight: 'bold'
                }}
                data-testid="span"
            >
                {props.title}
            </span>

            <Menu
                id="basic-menu"
                anchorEl={anchorEl}
                open={open}
                onClose={handleClose}
                MenuListProps={{
                    'aria-labelledby': 'basic-button',
                }}
            >
                {
                    props.menuItems.map((item, index) => <MenuItem 
                    key={index} 
                    onClick={handleClose}> <Link
                        style={{
                            "textDecoration": "none",
                            "color": "white",
                        }}
                        to={item.path}> {item.title} </Link> </MenuItem>)
                }

            </Menu>
        </div>
    );
}