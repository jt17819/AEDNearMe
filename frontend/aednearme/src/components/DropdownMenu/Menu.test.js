import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {MyMenu} from './Menu';
import axios from 'axios'


describe('Menu', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });



  // Rendering

  it('renders welcome message', () => {
    render(<MyMenu />, { wrapper: MemoryRouter })
    expect(screen.getByTestId('span')).toBeInTheDocument();
  });

//   test('list contains 1 buttons', () => {
//     render(<MyMenu />, { wrapper: MemoryRouter });
  
//     // const listElement = screen.getAllByRole('button');
//     const listItems = screen.getAllByRole('button');
  
    
//     expect(listItems.length).toEqual(1);
//   });



  // Snapshot

  it('should take a snapshot', () => {
    render(<MyMenu />, { wrapper: MemoryRouter })
    const { asFragment } = render(<MyMenu />, { wrapper: MemoryRouter })
    
    expect(asFragment(<MyMenu />)).toMatchSnapshot()
  })

  

  
})

