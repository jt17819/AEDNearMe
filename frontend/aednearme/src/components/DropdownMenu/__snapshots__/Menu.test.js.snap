// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <span
      aria-haspopup="true"
      data-testid="span"
      id="basic-button"
      style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
    />
  </div>
</DocumentFragment>
`;
